src/app/shared/components/recently-viewed/recently-viewed.component.html

```diff
@@ -0,0 +1,6 @@
+<app-heading *ngIf="products.length" [alertText]="'recently viewed'" [headingText]="'list'"></app-heading>
```

Comment:
recently viewed component should not render at all if products are absent. So please remove this check as you have it in the parent component